front/src/containers/PostSurveyForm.js

```diff
@@ -1,361 +1,470 @@
+      let roster = this.state.roster;
+      roster[index] = roundRoster;
+      this.setState({ roster: roster });
+    });
+    const actualPartnerName = singleTeam
+      ? singleTeamInfo.actualPartnerName
+      : "na";
+    let sOptions;
+    try {
+      sOptions = singleTeam
+        ? this.nicksFromRoster(
+            this.state.roster,
+            singleTeamInfo.roundsForSurvey[1]
+          ).map((x, ind, arr) => {
+            return {
+              value: `${x} ${actualPartnerName} ${arr.length}`,
+              label: x
+            };
+          })
+        : [{ value: "na", label: "na" }];
+    } catch (e) {
+      sOptions = [{ value: "na", label: "na" }];
     }
-}
-
-	// picks 2 random non-experimental rounds.
-	// takes 1 person from first round
-	// takes all persons from second round
-	// generates options from the second one
-	// returns {expPersonRound1: ... , sOptions: ... , actualPartnerName: ...}
-	generateSingleTeamInfo() {
-		const batch = this.props.batch;
-		if (batch.teamFormat !== 'single') {
-		  return null;
+    this.setState({
+      qOptions: qOptions,
+      uOptions: uOptions,
+      sOptions: sOptions
+    });
+    if (singleTeam) {
+      this.setState({
+        firstNick: singleTeamInfo.expPersonRound1Nick,
+        roundsForSurvey: singleTeamInfo.roundsForSurvey
+      });
     }
-		const userId = this.props.user._id.toString();
-		const [expRounds, worstRounds] = [batch.expRounds.map(x => x - 1), batch.worstRounds.map(x => x - 1)];
-		const numRounds = batch.numRounds;
-		const allRounds = Array.from(Array(numRounds).keys());
-		const roundPairs = batch.roundPairs;
+  }
 
-		let nonExpRounds = allRounds;
-		//Attempt to leverage batch variables
-		if(batch.dynamicTeamSize && batch.dynamicOptions){ //Team First [n,1,n,1]
-			nonExpRounds = nonExpRounds.filter((element, index) => index%2 === 0);
-		}else if(batch.dynamicTeamSize && !batch.dynamicOptions){ //Individual First [1,n,1,n]
-			nonExpRounds = nonExpRounds.filter((element, index) => index%2 !== 0);
-		}else{ //not using dynamic team sizing at all - should have the regular logic.
-			nonExpRounds = allRounds.filter(x => expRounds.indexOf(x) < 0 && worstRounds.indexOf(x) < 0);
-		}
+  // picks 2 random non-experimental rounds.
+  // takes 1 person from first round
+  // takes all persons from second round
+  // generates options from the second one
+  // returns {expPersonRound1: ... , sOptions: ... , actualPartnerName: ...}
+  generateSingleTeamInfo() {
+    const batch = this.props.batch;
+    if (batch.teamFormat !== "single") {
+      return null;
+    }
+    const userId = this.props.user._id.toString();
+    const [expRounds, worstRounds] = [
+      batch.expRounds.map(x => x - 1),
+      batch.worstRounds.map(x => x - 1)
+    ];
+    const numRounds = batch.numRounds;
+    const allRounds = Array.from(Array(numRounds).keys());
+    const roundPairs = batch.roundPairs;
 
-		const shuffledNonExpRounds = shuffle(nonExpRounds);
```

Comment:
@xehu I removed the shuffle functionality here. Maybe I can insert an if statement instead such that if your features are turned on, the tasks get shuffled? What do you think would be the best variable to track?

Reply:
Okay, double checked the post-survey functionality, and mine are not affected!